tensorboard/webapp/notification_center/_redux/testing.ts

```diff
@@ -33,3 +42,38 @@ export function buildStateFromNotificationState(
 ): State {
   return {[NOTIFICATION_FEATURE_KEY]: runsState};
 }
+
+@Injectable()
+export class TestingNotificationCenterDataSource
+  implements NotificationCenterDataSource {
+  fetchNotifications() {
+    return of({
+      notifications: [],
+      error: '',
+    });
+  }
+}
+
+export function provideTestingNotificationCenterDataSource() {
+  return [
+    TestingNotificationCenterDataSource,
+    {
+      provide: NotificationCenterDataSource,
+      useExisting: TestingNotificationCenterDataSource,
+    },
+  ];
+}
+
+export function buildDataSourceNotification(): DataSourceNotifications {
```

Comment:
this is not a builder: it does not accept any overrides or any capability to build a fixture. (1) Take an override, or (2) say `createNotification()`.